### PR TITLE
Provide persistent storage and data access classes

### PR DESCRIPTION
**Pull request by Nippleduck**
### Done
- created base domain entities 
- setup db context with entities configurations 
- migrated initial database schema
- added repositories with corresponding interfaces 
- added extension for DI

### About implementation
Usually in a project like this i would create repositories and unite them under one UnitOfWork class, but i am uncertain whether it is really necessary. 

**Why would i use:** 
- Single point of saving changes. Since we are using repositories seperately it is not always logically clear which one to use when saving chages in a situations when single method operates multiple repositories or changes multiple entities
- Save when needed. As alternative to the previous point we could save changes to database at the end of each modification method inside repositories. But in that case we cannot decide when to save changes which may lead to excessive db calls (especialy when we modify multiple tables and want to proceed with single transaction) 
Overall i think UOF is more compact and easier to expand when new repositories arrive rather then injecting them into constructors. For me it seems that whenever we use repositories UOF is a must(its either that or using context directly inside application services / use cases)

**The qustions is: is it really necessary and what would you personaly use or used in other projects?**

### Further development
In the next pull request i would like to implement skeleton for the rest projects inside solution and combine it with the first feature to speed up a little. Also which way is better to review new pull request: make them smaller or larger? There are several epics which contain multiple features to be implemented. So, would it be better to make pull request for each feature or epic(multiple features per epic)?

### CODE REVIEW

File: GameStore.Domain/Entities/Genre.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿using GameStore.Domain.Interfaces;
+
+namespace GameStore.Domain.Entities
+{
+    public class Genre : IEntity<Guid>
+    {
+        public Guid Id { get; set; }
+        public string? Name { get; set; }
+
+        public Guid? ParentGenreId { get; set; }
```
Comment (DMblTPO) [inline GameStore.Domain/Entities/Genre.cs:10]: GUID is the value type - structure, so, ParentGenreId as FK could be NULL?

File: GameStore.Persistence/Context/ApplicationDbContext.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+using System.Reflection;
+
+namespace GameStore.Persistence.Context
+{
+    public class ApplicationDbContext : IdentityDbContext<ApplicationUser>
+    {
+        public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options) 
+            : base(options) { }
+
+        public DbSet<Game> Games => Set<Game>();
```
Comment (DMblTPO) [inline GameStore.Persistence/Context/ApplicationDbContext.cs:14]: hmm... is it a new style to set dbset, instead of the usual get/set pair?

**Review by DMblTPO — APPROVED** [review]
ok

File: GameStore.Domain/Entities/Genre.cs (hunk 1 above)
Comment (Nippleduck) [reply GameStore.Domain/Entities/Genre.cs:10]: I made it nullable to notify entity framework that the ParentGenreId can be null (as there can be a genre with no parent, i.e. main genre)

File: GameStore.Persistence/Context/ApplicationDbContext.cs (hunk 2 above)
Comment (Nippleduck) [reply GameStore.Persistence/Context/ApplicationDbContext.cs:14]: I saw different approaches, guess they all work :)

**Pull request merged**